src/components/Editor/index.js

```diff
@@ -155,8 +154,10 @@ class Editor extends PureComponent {
       this.showSourceText(selectedSource, selectedLocation);
     }
 
-    if (this.props.outOfScopeLocations !== nextProps.outOfScopeLocations) {
-      clearLineClass(this.editor.codeMirror, "out-of-scope");
+    if (this.props.linesInScope !== nextProps.linesInScope) {
+      this.editor.codeMirror.operation(() => {
```

Comment:
`this.editor.codeMirror.operation` is a huge perf win. With that said, i'll be moving the operation inside `clearLineClass` soon :)